plugins/inputs/snmp/snmp.go

```diff
@@ -841,8 +799,131 @@ func fieldConvert(conv string, v interface{}) (interface{}, error) {
 	return nil, fmt.Errorf("invalid conversion type '%s'", conv)
 }
 
+type snmpTableCache struct {
+	mibName string
+	oidNum  string
+	oidText string
+	fields  []Field
+	err     error
+}
+
+var snmpTableCaches map[string]snmpTableCache
+var snmpTableCachesLock sync.Mutex
+
+// snmpTable resolves the given OID as a table, providing information about the
+// table and fields within.
+func snmpTable(oid string) (mibName string, oidNum string, oidText string, fields []Field, err error) {
```

Comment:
I'm unsure about the name of the function. `snmpTable` isn't reflective of the fact that it's calling out to the net-snmp `snmptable` utility. I just kept the pattern of the `snmpTranslate` function. But dunno that it's really that important.

Naming things is hard :-)